Utilities/Files.cs

```diff
@@ -75,10 +75,15 @@ public static void Write(string name, string content)
         {
             if (name != null && content != null)
             {
+                object fileLock = new object();
```

Comment:
Sorry, this won't help you: locking on a temporary stack object is meaningless since each entry into the function will get its own object to lock on, so will automatically pass! 

In the context of this class, the only thing you could lock on would be class member of the `Files` class, which would serialize **all** filesystem access. I don't think that's what we want either.

I think we need to step back and analyze where and how these IOExceptions are occurring. Any concurrency locking needs to happen higher up the call stack, I suspect.

Reply:
Unfortunately, I haven't been able to replicate the issue and I don't think I have enough information to say exactly how it is occurring. I can address the rest of it though.